### DISCUSSION TITLE
Making mass changes to the maintenance types and tasks in Device Model and Inventory items

### DISCUSSION THREAD

**Discussion by BioMedGPH** [Samedis-care/samedis-care-community#78] [Ideas]
Sometimes there is major revisions in the way things are set up in Samedis. It would be great if there was a way to force information that is defined the the Own Device Model or Own Device Type to be forced down to the inventory items of that Type/Model. Like the maintenance tasks. When changing maintenance types or maintenance frequencies/intervals in the Own Device Model  I would expect all inventory items of that Device Model to be reflecting that change and have these Maintenance types listed in the inventory item and even in de Maintenance Tasks if possible. This would make it possible to make bigger changes to the setup without having to manually process all 1000 inventory items after already processing 200 Device Models (or so). Maybe there can be some dialog in which you can select certain device model(s), and select what information you want to push to the dependent inventory items and and maintenance tasks that are New/Planned? Or when saving a change to a Device Model maybe the system can ask if you want this change to be applied to the depending inventory items?

Like we discussed via email, we are considering to change our 1 maintenance task per device type to having different maintenance types (quartely, semi-annual, annual, etc) and have these in 1 task and let the system figure out which type is to be done every 3/6/12 months. To make this change to our whole inventory would be a daunting task: 1) making the change in the Device Model, 2) applying the change in the service type settings of all the inventory items, 3) applying the change in all New/to do tasks for all these inventory items. I better get myself a full year's stock of coffee beans to get through that!

**Comment by BioMedGPH** [comment]
I did a quick test making an inventory item with 3 task types, quartely, semi-annual and annual. Created a task with all 3 types. Finished a task and let the system automatically find next maintenance date. It gave warning that multiple tasks were found and that it will choose the shortest interval (3 month). I then finished that task and again let it choose next task; again it choose 3 months. But nowhere I can indicate or keep track of whether during this task the quarterly, semi-annual or annual maintenance was completed. So with next task I have no idea which maintenance type is due. Maybe when completing you could indicate with a dropdown box what maintenance type was done, and as such the system can suggest which type to be done for the next maintenance?) In the task list view it now gives all 3 maintenance types as title. It would be better if it would show only 1 maintenance type in the title: the type that is due for that date.

**Comment by YvesR** [comment]
So we are discussing here two topics:
- Multiple processing changes via a wizard?
- Summarize tasks and set period instead of date so that date is automatically determined?

For massive changes there could be one way:
1. Select any items in the grid view
2. Click on "Batch-Change"
3. Select field to change
4. Set new values
5. Get Summary (456 inventories affected, changing maintenance interval A from 12 to 6 months) > Go Button

For the tasks it seems we would need to cover two things:
- Define series patterns, e.g. every 3 months.
- Maintenance types are included in the example 3, which include 3, 6 and 12. It would then need a display of what was done in the last. So you click on a checkbox at the end of a task, which interval has been processed and this is displayed in the next task, so that it is clear what needs to be done.
